### PR TITLE
Improve Webpack config

### PR DESCRIPTION
**Pull request by soupette**
#### Description of what you did:
Update the webpack config:
- `node-sass` has been removed
- `offline-plugin` has been added to cache some of the assets in production
- New cli option `strapi build --optimize` has been added. The idea is to reduce the time of the administration panel build process. 

#### My PR is a:

- [ ] 💥 Breaking change
- [ ] 🐛 Bug fix
- [x] 💅 Enhancement
- [ ] 🚀 New feature

#### Main update on the:

- [x] Admin
- [ ] Documentation
- [ ] Framework
- [ ] Plugin

#### Manual testing done on the following databases:

- [ ] Not applicable
- [ ] MongoDB
- [ ] MySQL
- [ ] Postgres
- [ ] SQLite

### CODE REVIEW

**Comment by alexandrebodin** [comment]
You should inverse the options it should build with optimization by default. and add a no-optimization option so you can use it in the yarn develop command when call the build script.

**Comment by soupette** [comment]
> You should inverse the options it should build with optimization by default. and add a no-optimization option so you can use it in the yarn develop command when call the build script.

Sure will do

File: docs/3.0.0-beta.x/cli/CLI.md [hunk 1]
```diff
@@ -69,6 +69,17 @@ By default the `development` environment will be used.
 
 Builds your admin panel.
 
+```bash
+strapi build
+
+options: [--optimize]
```
Comment (alexandrebodin) [inline docs/3.0.0-beta.x/cli/CLI.md]: update --no-optimization

File: packages/strapi-admin/admin/src/app.js [hunk 2]
```diff
@@ -195,3 +196,10 @@ export { dispatch };
 if (window.Cypress) {
   window.__store__ = Object.assign(window.__store__ || {}, { store });
 }
+
+// Install ServiceWorker and AppCache in the end since
+// it's not most important operation and if main code fails,
+// we do not want it installed
+if (NODE_ENV === 'production') {
+  require('offline-plugin/runtime').install(); // eslint-disable-line global-require
```
Comment (alexandrebodin) [inline packages/strapi-admin/admin/src/app.js]: adding service workers is kinda risky as of now it is not well known and used. we should be careful about that one.

File: packages/strapi-admin/package.json [hunk 3]
```diff
@@ -106,6 +106,7 @@
   "gitHead": "c85658a19b8fef0f3164c19693a45db305dc07a9",
   "devDependencies": {
     "chokidar": "^3.1.1",
+    "compression-webpack-plugin": "^3.0.0",
```
Comment (alexandrebodin) [inline packages/strapi-admin/package.json]: shouldn't be in dev-dep

Comment (alexandrebodin) [reply packages/strapi-admin/package.json]: neither shoud webpack and webpack-cli btw

File: packages/strapi-admin/webpack.config.js [hunk 4]
```diff
@@ -49,17 +72,16 @@ module.exports = ({
...
-          loader: MiniCssExtractPlugin.loader,
-          options: {
-            fallback: require.resolve('style-loader'),
-            publicPath: options.publicPath,
-          },
-        },
-      ]
-    : [{ loader: 'style-loader', options: {} }];
+  if (optimize && isProduction) {
+    webpackPlugins.push(
```
Comment (alexandrebodin) [inline packages/strapi-admin/webpack.config.js]: No need to used this. the files are already compressed by the server.

File: packages/strapi/lib/commands/build.js [hunk 5]
```diff
@@ -39,6 +39,7 @@ module.exports = async () => {
       dir,
       // front end build env is always production for now
       env: 'production',
+      optimize: optimization || false,
```
Comment (alexandrebodin) [inline packages/strapi/lib/commands/build.js]: || true

**Pull request merged**